File: palindrome.cs
using System;

class Test {
	static void Main () {
		Console.WriteLine (palindrom ("race cara") ? "palindrome" : "not palindrom");
	}
	
	public static bool palindrom (String arg) {
		arg = arg.Replace (" ", string.Empty).ToLower ();
		
		for (int i = 0, j = arg.Length - 1; i < j; ++i, --j)
			if (arg [i] != arg [j])
				return false;
		
		return true;
	}
}

File: Palindrome.java
public class Palindrome {
	public static void main (String [] args) {		
		if (palindrome ("race car"))
			System.out.println ("palindrome");
		else
			System.out.println ("not palindrome");
	}
	
	public static boolean palindrome (String arg) {
		arg = arg.replaceAll ("\\s+", "").toLowerCase ();
		
		for (int i = 0, j = arg.length () - 1; i < j; ++i, --j)
			if (arg.charAt (i) != arg.charAt (j))
				return false;
		
		return true;
	}
}